source/ID/concordium-did.rst

```diff
@@ -0,0 +1,479 @@
+The resulting DID is ``did:ccd:NET:pkc:PK`` where ``PK`` is the base58 encoded public key.
+These DIDs are not registered on the blockchain.
+
+Read
+----
+
+Account DID
+^^^^^^^^^^^
+
+The DID document information for a DID of the form
+
+``did:ccd:NET:acc:ADDR``
+
+can be resolved by looking up the account with address ``ADDR`` on blockchain ``NET``.
+
+Data required to construct the DID document can be acquired by using the gRPC interface command ``GetAccountInfo``.
+
+See the details in the `gRPC v2 documentation`_.
+
+From the command line, ``concordium-client`` allows to retrieve the data in the following way:
+
+.. code-block:: console
+
+    $concordium-client raw GetAccountInfo ADDR
+
+.. TODO add more details?
+
+
+Smart Contract Instance DID
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The DID document information for a DID of the form
+
+``did:ccd:NET:sci:IND:SUBIND``
+
+can be resolved by looking up the smart contract instance with indices ``IND``, ``SUBIND`` on blockchain ``NET``.
+This includes a lookup of the owner's account.
+
+Data required to construct the DID document can be acquired by using the gRPC interface command ``GetInstanceInfo``.
+
+See the details in the `gRPC v2 documentation`_.
+
+From the command line, ``concordium-client`` allows for retrieving the data in the following way:
+
+.. code-block:: console
+
+  $concordium-client contract show IND
+
+
+.. _sci-reference:
+
+Smart Contract Instance Reference
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+*Dereferencing* the smart contract DID reference invokes the specified entrypoint.
+
+Dereferencing a DID reference of the form
+
+``did:ccd:NET:sci:IND:SUBIND/EP?parameter_json=PAR``
+
+can be done by using the gRPC interface command ``InvokeInstance``.
+The entrypoint is considered a *view*: no state changes are persisted, only the result of the invocation is returned to the caller.
+The result of the invocation is the return value produced by the contract or an error, if the invocation failed.
+The return value is in the JSON format corresponding to the embedded smart contract schema.
+
+.. TODO should the binary return values be allowed? What if the contract doesn't have an embedded schema?
+
+From the command line, ``concordium-client`` allows for invoking a smart contract instance in the following way:
+
+.. code-block::
+
+  $concordium-client contract invoke IND --entrypoint EP --energy 3000000 --parameter-json param.json
+
+The base58 encoding of the ``param.json`` file corresponds to ``PAR``.
+
+See the details in the `gRPC v2 documentation`_.
+
+.. seealso::
+
+  `Dereferencing a DID URL`_ in the W3C Credentials Community Group draft report.
+
+.. TODO: write about binary vs JSON data
+
+Public Key Cryptography DID
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The DID document corresponding to a DID of the form
+
+``did:ccd:NET:pkc:PK``
+
+can be constructed directly from the DID without any lookup necessary.
+
+.. code-block:: json
```

Comment:
The code block doesn't seem very relevant. So, I would just remove it (unless it's giving some information about pkc-DID document that I cannot see?).

Reply:
Removed and added a note about the network